### ISSUE TITLE
PG 3.1 Can't upload new episodes.

### ISSUE THREAD

**Issue by R2D2C3POOU812** [PodcastGenerator/PodcastGenerator#283]
**System Information**
Podcast Generator Version: 3.1
Webserver: Litespeed 5.4.7 or Apache 2.4.43 
PHP Version: 7.2.30

**Podcast Generator Config**

```
<?php

$podcastgen_version = "3.1"; // Version

$first_installation = 1591576989;

$installationKey = "CENSORED";

$scriptlang = "en_US";

$url = "https://podcast.xxx.ca/PodcastGenerator-3.1/PodcastGenerator/";

$absoluteurl = "/home/xxx/xxx.ca/PodcastGenerator-3.1/PodcastGenerator/"; // The location on the server

$theme_path = "themes/default/";

$upload_dir = "media/"; // "media/" the default folder (Trailing slash required). Set chmod 755

$img_dir = "images/"; // (Trailing slash required). Set chmod 755

$feed_dir = ""; // Where to create feed.xml (empty value = root directory). Set chmod 755

$max_recent = "all"; // How many file to show in the home page

$recent_episode_in_feed = "All"; // How many file to show in the XML feed (1,2,5 etc.. or "All")

$episodeperpage = 10;

$enablestreaming = "yes"; // Enable mp3 streaming? ("yes" or "no")

$freebox = "yes"; // enable freely customizable box

$enablepgnewsinadmin = "yes";

$strictfilenamepolicy = "no"; // strictly rename files (just characters A to Z and numbers) 

$categoriesenabled = "yes";

$cronAutoIndex = 1; //Auto Index New Episodes via Cron

$cronAutoRegenerateRSS = 1; //Auto regenerate RSS via Cron

$indexfile = "index.php";    // Path of the index file

$podcastPassword = "";       // Password to protect the podcast generator webpages, this will NOT protect the audio or XML files. Leave blank to disable.

#####################

# XML Feed stuff

$podcast_title = "THIS SECTION IS CENSOREDt";

$podcast_subtitle = "THIS SECTION IS CENSORED";

$podcast_description = "THIS SECTION IS CENSORED";

$author_name = "THIS SECTION IS CENSORED";

$author_email = "THIS SECTION IS CENSORED";

$itunes_category[0] = "THIS SECTION IS CENSORED"; // iTunes categories (mainCategory:subcategory)

$itunes_category[1] = "THIS SECTION IS CENSORED";

$itunes_category[2] = "THIS SECTION IS CENSORED";

$link = "?name="; // permalink URL of single episode (appears in the <link> and <guid> tags in the feed)

$feed_language = "en";

$copyright = "All rights reserved";   // Your copyright notice (e.g CC-BY)

$feed_encoding = "utf-8";

$explicit_podcast = "no"; //does your podcast contain explicit language? ("yes" or "no")

$users_json = "{\"THIS SECTION IS CENSORED\": \"\$2y\$10\$BbBA5.MGzlzbhA5ukXu8v.0JTWspZM2Wyo6CVsTITNfRBQw/Ufou2\"}";

// END OF CONFIG


INSERT THE **CENSORED** CONFIG HERE
``` 
How and where do I find the original PG config file? Is it in the ...PodcastGenerator/admin?
I see a file there called **pg_config.php** Is this what I'm looking for? 

*Edit* I believe I found the config file after comparing it to other user issues. 
Like others, the tool at the link above was there, however, it did not work after copying and pasting the details into the box and pressing "evaluate". Should I have dragged the entire config file into the window? Either way, I edited it manually. If there is anything I missed please let me know.

**Side-Note** I noticed that my config file is missing the **$username and $userpassword** scripts. I did not edit it out, it just wasn't there, to begin with. Is this causing an issue perhaps?

**Description of your problem:**

To clarify from above I am the admin on my hosting plan and spoke to a tech rep who told me that they are using Litespeed as their webserver yet when I go into server information it shows that's its using apache v.2.4.43. Can it be using both simultaneously?

I'm not sure if it matters pertaining to my issue.

I installed v.3.1 on a subdomain of my web hosting platform. I used FileZilla to upload the unzipped folder directly into the subdomain.
I was able to launch the set up by pointing URL to the correct folder from the cPanel File Manager.
I created a password and went to upload a new episode using mp3.

Ther error occurs when I try to upload a new episode.
I can't seem to finish the process. It shows percentage uploading in the bottom left side of the browser. Then when it gets close to 100% an error says "The upload was not successfully"
It's always the same error but initially, I got some kind of category error.

I always fill out all the fields properly. Are there restrictions on file types or naming files? I ended up renaming mp3 with one word to eliminate underscores, spaces, and any characters. My filename was **Once.mp3** So I don't think that was the problem.

Are there any restrictions on short or long description fields. Title Fields?

On a side note, I don't understand the difference between the category drop-down menu and the manage categories where you can manually edit the category names? I had to create categories and they are present when I try to upload new episode but there's no confirmation that they are selected. They just highlight blue when you click on them.

I hope this makes sense. 

I'm very new to coding so guide me as best as you can.

**Comment by ryangurn** [comment]
It is possible that the webserver is setup with both litespeed and apache! This _**could**_ be causing the issues with uploading. To confirm my understanding it appears that the installation goes through successfully for PG3.1 however when you are attempting to upload the episodes it will get to 100% and then not succeed, is that correct?

**Question:** Can you confirm if the deployment of the server is currently using the litespeed webserver or if it is in-fact using apache. Just so you know litespeed can use apache configurations so I just want to be sure that its not a version variable that is set incorrectly in the httpd.conf

If you want to see litespeeds feature page it describes this, and I have provided a link that will highlight the text that I am discussing.

[Litespeed Feature Page](https://www.litespeedtech.com/products/litespeed-web-server/features/apache-drop-in-replacement#:~:text=LiteSpeed%20Web%20Server%20is%20Apache%20compatible&text=This%20means%20there's%20no%20configuration,control%20panels%20written%20for%20Apache.)

**Comment by R2D2C3POOU812** [comment]
Hey Ryan,

Thanks for getting back to me.

I spoke with tech support at my host and he confirmed that I am in fact using Litespeed and Apache simultaneously.  As they described on the Litespeed website from the link you provided this is possible and Apache _is_ running underneath.

I don't suppose I can ask them to change this as it probably is how they have all their servers set up.

Sidenote: Tried to reinstall software through softaculous as it now displays the current version as 3.1 updated in the last week or. I installed it successfully but couldn't set up a password. Every time I tried to log in with the password and username I created it would say invalid. 

I also noticed that through softaculous the install process is a little different than the 3 step process when upload files through FTP and pointing URL to the current folder to start the  set up. 

I ended up using FileZilla to reupload to the subdomain and it worked, however when I got to uploading mp3 ran into the same issue as before, "file was not uploaded successfully".

**Comment by ryangurn** [comment]
Ok good to know, I will look into the codebase and see if I can identify anything. 

Recap of issues:
1. Admin password not functional when using softaculous
2. File not uploaded successfully error message each time

By chance did you attempt to use the FTP method of adding in MP3’s, did that function properly?

**Comment by ryangurn** [comment]
@R2D2C3POOU812 Additionally, can you talk a look at issue #270 since it appears that it might help answer some questions about the password mismatch.

**Comment by R2D2C3POOU812** [comment]
Aha! I found out what the issue was. It was none of the above unfortunately and I only thought of it because after getting if the phone with a tech support rep at my hosting co. I realized something. 

When I first installed Wordpress on my server way back I was having trouble uploading files larger than 64mb and after much searching found out that I needed to change the upload limit size in the theme functions.php file to higher, something like 512mb. 

After navigating around in my cPanel, in the multiPHPini editor, I noticed the maximum file size upload was set to 2mb so I changed it to 2gb. After saving the changes I was able to upload the audio file, however, only after changing it from m4a to mp3. There was an error on PG3.1 saying unsupported MIME type? I thought m4a or m4v were ok for a podcast?

I submitted to apple after checking with https://castfeedvalidator.com/ as I was having errors reporting **invalid categories** and **incorrect artwork dimensions**. I made changes to the image in photoshop reuploaded and apple approved podcast this morning so I'm up and running.

*side note - PG generator only accepts jpeg? I tried using png but I got an error saying "invalid  file format" or something similar

I did however have some obstacles: When uploading new artwork image, the file would not take, and the default image would still be there. Instead of using the PG interface, I uploaded the file using Filezilla and the file took, except the feed.xml would not reflect the update even though I could see using file manager that the new image was there. Even after regenerating RSS it wouldn't change in the .XML file.

 I ended up manually editing that file to reflect changes and after that the errors on castfeedvalidator cleared up. 

**Recap**

So there were lots of small obstacles but the big one being the php maximum file size on the server.

To answer some other questions:

 I uploaded 3.0 using FTP (FileZilla) originally as I knew I had to update from there then updated from that to 3.01. Eventually, I just removed those from the server and uploaded 3.1 to subdomain without upgrading and pointed the URL to podcastgenerator on my server wherein I was able to go through the 3 step process wizard and create password. 

When I got the mp3 uploaded I did not use FTP for that just the PG3.1 interface.
I suppose in hindsight I might have been able to do it this way but would the cPanel PHP maximum file size still not allow it? I don't know how that works. By uploading through Filezilla could I have bypassed by my PHP server limits If I uploaded directly into the appropriate folder?

The issue with softaculous was that after install, it displays a link that is slightly different from the URL that creates the 3 step wizard install such as https://xxx.xxx.ca/podcastgenerator/admin.php
(not exactly this but close, I can't remember the URL. The actual URL to create the 3 step process just drops the /admin and ends with podcastgenerator/ or whereever your actual files are located. 

I hope this makes sense. I'm sorry for confusing anyone but I's so happy it works now.

I have some other questions regarding the app, but are not related to the initial problem I had here. Should I start a new thread for that?

Also sorry for the novel
Thank you both Ryan and Emil for tackling this together.

**Comment by emilengler** [comment]
> After navigating around in my cPanel, in the multiPHPini editor, I noticed the maximum file size upload was set to 2mb so I changed it to 2gb. After saving the changes I was able to upload the audio file, however, only after changing it from m4a to mp3. There was an error on PG3.1 saying unsupported MIME type? I thought m4a or m4v were ok for a podcast?

Good to know, the web server is in 90% of the cases the troublemaker

> *side note - PG generator only accepts jpeg? I tried using png but I got an error saying "invalid file format" or something similar

Yes, but this is intended to be that way because iTunes only supports JPEG files (maybe it was changed nowadays)

> By uploading through Filezilla could I have bypassed by my PHP server limits If I uploaded directly into the appropriate folder?

Yes

**Comment by emilengler** [comment]
Happy Podcasting!

**Comment by thadeu220** [comment]
i can't upload new episodes... it says this:
Unsupported mime type detected for file with extension .m4a

how do i fix this? i need to upload as M4a / AAC...
i sawd something about adding a new mime type via litespeed, using htaccess... but no idea where do i put the moded .htaccess file

do i put in the pg 3.1 directory, or, in the root of my /home/user folder?

help

**Comment by coldacid** [comment]
@thadeu220 your particular issue is not related to htaccess or litespeed httpd but rather a component of Podcast Generator itself. This is fixed in 3.1.2 but that version itself has problems caused when upgrading from an earlier version of PG.

For now, [download this file](https://raw.githubusercontent.com/PodcastGenerator/PodcastGenerator/master/PodcastGenerator/components/supported_media/supported_media.xml) and copy it over components/supported_media/supported_media.xml in your Podcast Generator directory. That should fix the M4A/AAC upload problem.

**Comment by thadeu220** [comment]
New release has fixed the bug
Thanks
